test_conformance/math_brute_force/CMakeLists.txt

```diff
@@ -15,28 +17,37 @@ set(${MODULE_NAME}_SOURCES
     function_list.h
     i_unary_double.cpp
     i_unary_float.cpp
+    i_unary_half.cpp
     macro_binary_double.cpp
     macro_binary_float.cpp
+    macro_binary_half.cpp
     macro_unary_double.cpp
     macro_unary_float.cpp
+    macro_unary_half.cpp
     mad_double.cpp
     mad_float.cpp
+    mad_half.cpp
     main.cpp
     reference_math.cpp
     reference_math.h
     sleep.cpp
     sleep.h
     ternary_double.cpp
     ternary_float.cpp
+    ternary_half.cpp
     test_functions.h
     unary_double.cpp
     unary_float.cpp
+    unary_half.cpp
```

Comment:
Could you verify that the kernels that are added that use the half type have the pragma for enabling cl_khr_fp16:
#pragma OPENCL EXTENSION cl_khr_fp16 : enable"

Reply:
I just run all the `bruteforce` tests related to `cl_khr_fp16` and yup - I can confirm all kernels contain necessary `pragma`, roughly header of each kernel looks very similar:

```
#define RETTYPE half2
#define RETTYPE_SCALAR half
#define TYPE1 half2
#define TYPE1_SCALAR half
#define UNDEF1 NAN
#pragma OPENCL EXTENSION cl_khr_fp16 : enable

__kernel void math_kernel2(__global RETTYPE* ....
```

Reply:
This is the kernel that's getting generated for nan:

    Build options:  -cl-std=CL3.0
     Original source is: ------------
     #define RETTYPE half16
     #define RETTYPE_SCALAR half
     #define TYPE1 ushort16
     #define TYPE1_SCALAR ushort
     #define UNDEF1 0x5678
     
     __kernel void math_kernel16(__global RETTYPE* out,
                               __global TYPE1* in1)
     {
         size_t i = get_global_id(0);
         out[i] = nan(in1[i]);
     }

Can you specifically check unary_u_half.cpp?

Reply:
Good catch, this is actually a pre-existing issue in `common.cpp`.  I've opened #1705 to fix this for fp64.  That fix will then have to be extended for fp16.

Reply:
Just to comprehend the problem - why I can't see this ? Is it specifically related to ARM architecture build ?

Reply:
Answering to myself - for some reason missing `pragma` did not cause neither kernel built error nor test fail with my Intel device.

Reply:
@shajder #1705 has been merged now, so if you rebase this PR on top of latest `main` then the missing pragma issue should be fixed.

Reply:
@shajder yes you will need to add `ParameterType::Short`/`UShort` to the switch (which is now inside a loop so it's one level deeper compared to your previous patch).  Then you also need to update the case for `ParameterType::Half` to reflect how `Double` is handled now.

Reply:
Correction merged, `nan` kernel now looks correct:

  _2:             nan fp16     #define RETTYPE half_

```
#define RETTYPE_SCALAR half
#define TYPE1 ushort3
#define TYPE1_SCALAR ushort
#define UNDEF1 0x5678
#pragma OPENCL EXTENSION cl_khr_fp16 : enable

__kernel void math_kernel3(__global RETTYPE_SCALAR* out,
                          __global TYPE1_SCALAR* in1)
{
    size_t i = get_global_id(0);
...
```